client/question/domain/PreprocessedCodeObjectFactory.js

```diff
@@ -0,0 +1,88 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of
+ * PreprocessedCode domain objects.
+ */
+
+tie.factory('PreprocessedCodeObjectFactory', [
+  function() {
+    /**
+     * PreprocessedCode stores the entire preprocessed code as well as the
+     * string separator generated during print output. This string separator
+     * is utilized in splitting the total output into the corresponding
+     * outputs for each test case.
+     */
+
+    /**
+     * Constructor for PreprocessedCode
+     *
+     * @param {string} preprocessedCode Preprocessed student code
```

Comment:
Maybe we should call this preprocessedCodeString, to distinguish it from the PreprocessedCode object that contains it.